recipes-bsp/u-boot/files/0011-board-siemens-iot2050-get-and-set-m.2-releated-infor.patch

```diff
@@ -0,0 +1,229 @@
+From ba758a41c49507b4abfcf292b695582f8cbb4375 Mon Sep 17 00:00:00 2001
+From: chao zeng <chao.zeng@siemens.com>
+Date: Fri, 7 Jan 2022 16:35:43 +0800
+Subject: [PATCH 11/16] board:siemens:iot2050:get and set m.2 releated
+ information
+
+Get the M.2 configuration,enable the power of the m.2
+Set the releated overlay according to the M.2 configuration
+
+Signed-off-by: chao zeng <chao.zeng@siemens.com>
+---
+ board/siemens/iot2050/board.c | 173 +++++++++++++++++++++++++++++++++-
+ 1 file changed, 171 insertions(+), 2 deletions(-)
+
+diff --git a/board/siemens/iot2050/board.c b/board/siemens/iot2050/board.c
+index 817fbf4ecb..21c82556e9 100644
+--- a/board/siemens/iot2050/board.c
++++ b/board/siemens/iot2050/board.c
+@@ -47,6 +47,141 @@ struct iot2050_info {
+ 
+ DECLARE_GLOBAL_DATA_PTR;
+ 
++#define M2_CONFIG_LEN    4
++#define PCIE_MUX_CONTROL 3
++#define SERDES_SELECT    2
++
++typedef struct gpio_config {
++	char *label;
```

Comment:
Use MACRO instead of string to save some space.

Reply:
for this context, we need to use string